### DISCUSSION TITLE
422 unprocessable entity fastapi when trying to call api to get image

### DISCUSSION THREAD

**Discussion by dolong2110** [fastapi/fastapi#8884] [Questions] [answered]
I am a newbie to fastAPI. While I tried to write an API to get the uploaded image I got this error: 
INFO:     127.0.0.1:50702 - "POST /faces/identify HTTP/1.1" 422 Unprocessable Entity

my codes here. It very thankful if someone can help

```
router = APIRouter()
# Identify faces on the image
@router.post('/faces/identify')
async def web_recognize(file_upload: UploadFile = File(...)):
    face_recognition_service = faceRecognition.FaceRecognition_service(faces_dict)
    file = face_recognition_service.extract_image(file_upload)
    if file and face_recognition_service.is_picture(file.filename):

        return face_recognition_service.detect_faces_in_image(file)
    else:
        raise ErrorResponse(code=codes.code_files_invalid, message=messages.message_files_invalid, `status=status.HTTP_400_BAD_REQUEST)`

```

**Comment by NiharZanwar** [comment, marked as answer]
A response having status code 422 will have a response body that specifies the error message, please have a look at that. 
Also please mention from where are you making the request.

Below example code should work
```
import requests
url = "http://localhost/faces/identify"
files = {'file_upload': open('test.jpg', 'rb')}
requests.post(url, files=files)
```

Basically, you will have to make a POST request having content-type: `multipart/form-data`

**Comment by dolong2110** [comment]
Dear here is my response
`{
    "detail": [
        {
            "loc": [
                "body",
                "image"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}`

and here is my curl in postman
`curl --location --request POST 'http://localhost:8080/faces/identify' \
--form 'file=@"1.jpg"'`

where the image is in working directory. I just put it there to test my api

**Comment by NiharZanwar** [comment]
Try `curl --location --request POST 'http://localhost:8080/faces/identify' \ --form 'file_upload=@"1.jpg"'`

**Comment by DMLON** [comment]
Hello, i'm having the same issue when passing an image with request. Have you been able to fix this?

**Comment by dolong2110** [comment]
I solved this my new curl is 
curl --location --request POST 'http://localhost:8080/faces/identify' \
--form 'images=@"2.jpg"'
You should import the image into postman and use this

**Comment by DMLON** [comment]
Mmmm, i need to do it with requests due to how my app works.

**Comment by dolong2110** [comment]
let me see the code when you get the image?

**Comment by DMLON** [comment]
I managed to make it work with requests, i followed this issue here:
#579 
Had to use starlette's requests and responses, here is my code if anyone runs into this issue

-- API
```python
@router.post('/detect_vehicle_bytes',response_class=Response)

async def detect_vehicle(data: Request):
    data_b = await data.body()
    result = vehicle_detector.detect(data_b)
    return JSONResponse(result)`
```

-- Client
```python
img_bytes = cv2.imencode('.jpg',image_input)[1].tobytes()
data = BytesIO(img_bytes)
response=requests.post(self.endpoint,data=data)
result = response.json()
```

Thanks!

**Reply by Dmitri-Sintsov** [reply]
I had similar 422 unprocessable entity error when tried to perform POST with JSON body to FastAPI route via built-in [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).

Switched to [axios](https://github.com/axios/axios) and POST with JSON with the same data is now performed successfully.

There is something picky in FastAPI algorithm of processing / validating JSON request body, I guess.

**Comment by dolong2110** [comment]
what is data_b, is it image type? or just json and what you do in vehicale_detector is parsing this json?
